src/test/java/marquez/common/models/UrnPatternTest.java

```diff
@@ -0,0 +1,116 @@
+  private static final int ALLOWED_PART_SIZE = 64;
+  private static final int PART_SIZE_GREATER_THAN_ALLOWED = ALLOWED_PART_SIZE + 1;
+  private static final String NAMESPACE = "test";
+  private static final int NUM_OF_PARTS = 2;
+  private static final UrnPattern PATTERN = UrnPattern.from(NAMESPACE, NUM_OF_PARTS);
+
+  @Test(expected = NullPointerException.class)
+  public void testFrom_throwsException_onNullNamespace() {
+    final String nullNamespace = null;
+    UrnPattern.from(nullNamespace, NUM_OF_PARTS);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testFrom_throwsException_onEmptyNamespace() {
+    final String emptyNamespace = "";
+    UrnPattern.from(emptyNamespace, NUM_OF_PARTS);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testFrom_throwsException_onBlankNamespace() {
+    final String blankNamespace = " ";
+    UrnPattern.from(blankNamespace, NUM_OF_PARTS);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testFrom_throwsException_onNullNumOfParts() {
+    final Integer nullNumOfParts = null;
+    UrnPattern.from(NAMESPACE, nullNumOfParts);
+  }
+
+  @Test
+  public void testThrowIfNoMatch_noExceptionThrown() {
+    final String value = "urn:test:a:b";
+    PATTERN.throwIfNoMatch(value);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testThrowIfNoMatch_throwsException_onNoMatchValue() {
+    final String noMatchValue = "urn:other:a:b";
+    PATTERN.throwIfNoMatch(noMatchValue);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testThrowIfNoMatch_throwsException_onNoPrefixValue() {
+    final String noPrefixValue = "other:a:b";
+    PATTERN.throwIfNoMatch(noPrefixValue);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNewDatasetUrn_throwsException_onMissingPartValue() {
+    final String missingPartValue = "urn:test:a";
+    PATTERN.throwIfNoMatch(missingPartValue);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testThrowIfNoMatch_throwsException_onExtraPartValue() {
+    final String extraPartValue = "urn:test:a:b:c";
+    PATTERN.throwIfNoMatch(extraPartValue);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testThrowIfNoMatch_throwsException_onNonAlphanumericPartValue() {
+    final String nonAlphanumericPartValue = "urn:test:a:b$c^";
+    PATTERN.throwIfNoMatch(nonAlphanumericPartValue);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testThrowIfNoMatch_throwsException_onPartGreaterThan64Value() {
+    final String partGreaterThan64Value = newUrnWithPartGreaterThan64();
+    PATTERN.throwIfNoMatch(partGreaterThan64Value);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testThrowIfNoMatch_throwsException_onNullValue() {
+    final String nullValue = null;
+    PATTERN.throwIfNoMatch(nullValue);
+  }
+
+  private String newUrnWithPartGreaterThan64() {
```

Comment:
To promote a one-stop-shop for generating test members, this can be moved to the `Generator` class.

Reply:
`newUrnWithPartGreaterThan64()` is strictly for testing the Urn class and probably not something that we'd want in **class** `Generator`.

We'll eventually want to revisit `Generator` usage to clean up a few things. Might be worth having the conversation then on where this method should live.